File: app/src/test/java/ru/iandreyshev/parserrss/models/repository/RssTest.java
package ru.iandreyshev.parserrss.models.repository;

import org.junit.Before;
import org.junit.Test;

import java.util.ArrayList;
import java.util.List;

import static org.junit.Assert.*;

public class RssTest {
    private static final String TITLE = "Title";
    private static final String ORIGIN = "Origin";
    private static final String DESCRIPTION = "Description";
    private static final String URL = "URL";
    private static final Article ARTICLE = new Article();
    private static final long ID = 12;
    private Rss mRss;
    private Rss mDefaultRss;

    @Before
    public void setup() {
        mRss = new Rss(TITLE, ORIGIN);
        mDefaultRss = new Rss();
    }

    @Test
    public void haveConstructorWithZeroArgs() {
        final Rss rss = new Rss();

        assertNotNull(rss);
    }

    @Test
    public void haveIdGetterAndSetter() {
        mRss.setId(ID);

        assertEquals(mRss.getId(), ID);
    }

    @Test
    public void haveTitleGetter() {
        assertEquals(mRss.getTitle(), TITLE);
    }

    @Test
    public void haveDescriptionGetterAndSetter() {
        mRss.setDescription(DESCRIPTION);

        assertEquals(mRss.getDescription(), DESCRIPTION);
    }

    @Test
    public void haveEmptyTitleByDefault() {
        assertEquals(mDefaultRss.getTitle(), "");
    }

    @Test
    public void haveEmptyUrlByDefault() {
        assertEquals(mDefaultRss.getUrl(), "");
    }

    @Test
    public void haveEmptyOriginByDefault() {
        assertEquals(mDefaultRss.getOrigin(), "");
    }

    @Test
    public void haveEmptyListOfArticlesByDefault() {
        assertEquals(mDefaultRss.getArticles().size(), 0);
    }

    @Test
    public void haveEmptyListOfArticlesViewsByDefault() {
        assertTrue(mDefaultRss.getArticles().isEmpty());
    }

    @Test
    public void haveArticlesGetterAndSetter() {
        final List<Article> articles = new ArrayList<>();
        articles.add(ARTICLE);
        mRss.setArticles(articles);

        assertEquals(mRss.getArticles().get(0), ARTICLE);
    }

    @Test
    public void haveUrlGetterAndSetter() {
        mRss.setUrl(URL);

        assertEquals(mRss.getUrl(), URL);
    }

    @Test
    public void hashcodeEqualsUrlHashcode() {
        mRss.setUrl(URL);

        assertEquals(mRss.hashCode(), URL.hashCode());
    }

    @Test
    public void equalsWithObjectIfEqualsItsUrls() {
        mRss.setUrl(URL);
        final Rss otherRss = new Rss();

        assertFalse(mRss.equals(otherRss));

        otherRss.setUrl(URL);

        assertTrue(mRss.equals(otherRss));
    }
}

File: app/src/test/kotlin/ru/iandreyshev/parserrss/models/repository/RssTest.kt
package ru.iandreyshev.parserrss.models.repository

import org.junit.Before
import org.junit.Test

import java.util.ArrayList

import org.junit.Assert.*

class RssTest {

    companion object {
        private const val TITLE = "Title"
        private const val ORIGIN = "Origin"
        private const val DESCRIPTION = "Description"
        private const val URL = "URL"
        private const val ID: Long = 12
        private val ARTICLE = ArticleEntity(title = TITLE, description = DESCRIPTION, originUrl = ORIGIN)
    }

    private lateinit var mRss: RssEntity
    private lateinit var mDefaultRss: RssEntity

    @Before
    fun setup() {
        mRss = RssEntity(title = TITLE, originUrl = ORIGIN)
        mDefaultRss = RssEntity(title = "", originUrl = "")
    }

    @Test
    fun haveIdGetterAndSetter() {
        mRss.id = ID

        assertEquals(mRss.id, ID)
    }

    @Test
    fun haveTitleGetter() {
        assertEquals(mRss.title, TITLE)
    }

    @Test
    fun haveDescriptionGetterAndSetter() {
        mRss.description = DESCRIPTION

        assertEquals(mRss.description, DESCRIPTION)
    }

    @Test
    fun haveEmptyTitleByDefault() {
        assertEquals(mDefaultRss.title, "")
    }

    @Test
    fun haveEmptyUrlByDefault() {
        assertEquals(mDefaultRss.url, "")
    }

    @Test
    fun haveEmptyOriginByDefault() {
        assertEquals(mDefaultRss.originUrl, "")
    }

    @Test
    fun haveEmptyListOfArticlesByDefault() {
        assertEquals(mDefaultRss.articles.size.toLong(), 0)
    }

    @Test
    fun haveEmptyListOfArticlesViewsByDefault() {
        assertTrue(mDefaultRss.articles.isEmpty())
    }

    @Test
    fun haveArticlesGetterAndSetter() {
        val articles = ArrayList<ArticleEntity>()
        articles.add(ARTICLE)
        mRss.articles = articles

        assertEquals(mRss.articles[0], ARTICLE)
    }

    @Test
    fun haveUrlGetterAndSetter() {
        mRss.url = URL

        assertEquals(mRss.url, URL)
    }

    @Test
    fun hashcodeEqualsUrlHashcode() {
        mRss.url = URL

        assertEquals(mRss.hashCode().toLong(), URL.hashCode().toLong())
    }

    @Test
    fun equalsWithObjectIfEqualsItsUrls() {
        mRss.url = URL
        val otherRss = RssEntity(title = "", originUrl = "")

        assertFalse(mRss == otherRss)

        otherRss.url = URL

        assertTrue(mRss == otherRss)
    }
}